hl/test/test_file_image.c

```diff
@@ -506,6 +506,8 @@ test_file_image(size_t open_images, size_t nflags, unsigned *flags)
     } /* end for */
 
     /* release temporary working buffers */
+    for (i = 0; i < open_images; i++)
```

Comment:
You may consider protecting all of these.

Reply:
The C standard says that freeing a NULL pointer is fine so checking for NULL isn't necessary.